src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -779,13 +855,24 @@ export class SearchView extends ViewPane {
 		e.browserEvent.preventDefault();
 		e.browserEvent.stopPropagation();
 
+		const match = e.element;
+		let context: IRenderableMatchContext | undefined = undefined;
+		if (match instanceof FileMatch) {
+			context = toFileMatchContext(match);
+		} else if (match instanceof Match) {
+			context = toMatchContext(match);
+		} else if (match instanceof FolderMatch) {
+			context = toFolderMatchContext(match);
+			context.resource = match.resource;
+		}
+
 		const actions: IAction[] = [];
 		const actionsDisposable = createAndFillInContextMenuActions(this.contextMenu, { shouldForwardArgs: true }, actions);
 
 		this.contextMenuService.showContextMenu({
 			getAnchor: () => e.anchor,
 			getActions: () => actions,
-			getActionsContext: () => e.element,
+			getActionsContext: () => context || e.element,
```

Comment:
Will `context` be undefined here? Also I wonder whether this should be an array, because we will want to enable multiselect in this list, easier to do that now.

Reply:
Changed